lib/rules/no-useless-set-operand.ts

```diff
@@ -0,0 +1,242 @@
+import type { RegExpVisitor } from "@eslint-community/regexpp/visitor"
+import type {
+    CharacterClassElement,
+    ClassSetOperand,
+    ExpressionCharacterClass,
+    Node,
+    StringAlternative,
+} from "@eslint-community/regexpp/ast"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor } from "../utils"
+import { toUnicodeSet } from "regexp-ast-analysis"
+
+type FlatElement = CharacterClassElement | StringAlternative
+
+function getFlatElements(
+    node: ClassSetOperand | ExpressionCharacterClass["expression"],
+): readonly FlatElement[] {
+    if (node.type === "ClassStringDisjunction") {
+        return node.alternatives
+    }
+    if (node.type === "CharacterClass") {
+        const nested: FlatElement[] = []
+        // eslint-disable-next-line func-style -- x
+        const addElement = (element: CharacterClassElement) => {
+            if (element.type === "ClassStringDisjunction") {
+                nested.push(...element.alternatives)
+            } else if (element.type === "CharacterClass") {
+                if (!element.negate) {
+                    nested.push(...element.elements)
+                }
+                nested.push(element)
+            } else {
+                nested.push(element)
+            }
+        }
+        node.elements.forEach(addElement)
+        return nested
+    }
+
+    return []
+}
+
+function removeDescendant(root: Node, e: FlatElement): string {
+    let { start, end } = e
+
+    if (e.type === "StringAlternative") {
+        if (e.parent.alternatives.length === 1) {
+            // we have to remove the whole string disjunction
+            // eslint-disable-next-line no-param-reassign -- x
```

Comment:
If you wish to turn off this rule for the entire repository, we can do so.
The ruleset is primarily intended for repositories that I maintain alone.

Reply:
No, I actually like this rule. It enforces that parameters are `const`, which I think is a good idea. It's just that sometimes (rarely) I need parameters to be `let`.